com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/detect/process/MtaMetadataExtractorFactoryImpl.java

```diff
@@ -0,0 +1,25 @@
+package com.sap.cloud.lm.sl.cf.core.cf.detect.process;
+
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.core.cf.detect.MtaMetadataExtractor;
+import com.sap.cloud.lm.sl.cf.core.cf.detect.MtaMetadataExtractorFactory;
+import com.sap.cloud.lm.sl.cf.core.cf.detect.entity.ApplicationMetadataEntity;
+import com.sap.cloud.lm.sl.cf.core.cf.detect.entity.MetadataEntity;
+import com.sap.cloud.lm.sl.cf.core.cf.detect.entity.ServiceMetadataEntity;
+
+@Component
+public class MtaMetadataExtractorFactoryImpl<T extends MetadataEntity> implements MtaMetadataExtractorFactory<T> {
+
+    @Override
+    public MtaMetadataExtractor<T> get(T entity) {
+        if(entity instanceof ApplicationMetadataEntity) {
+            return (MtaMetadataExtractor<T>) new AppMtaMetadataExtractor();
+        }
+        if(entity instanceof ServiceMetadataEntity) {
+            return (MtaMetadataExtractor<T>) new ServiceMtaMetadataExtractor();
+        }
+        return null;
```

Comment:
How is this handled?
Shouldn't we throw an IllegalStateException because of the unknown entity?